gtfs-realtime/spec/en/reference.md

```diff
@@ -190,6 +192,19 @@ The relation between this StopTime and the static schedule.
 | **SKIPPED** | The stop is skipped, i.e., the vehicle will not stop at this stop. Arrival and departure are optional. |
 | **NO_DATA** | No data is given for this stop. It indicates that there is no realtime information available. When set NO_DATA is propagated through subsequent stops so this is the recommended way of specifying from which stop you do not have realtime information. When NO_DATA is set neither arrival nor departure should be supplied. |
 
+## _enum_ WheelchairAccessible
+
+If a particuliar trip is accessible to wheelchair. When available, this value should overwrite the _wheelchair_accessible_ value from the static GTFS. 
+
+#### Values
+
+| _**Value**_ | _**Comment**_ |
+|-------------|---------------|
+| **NO_VALUE** | The trip doesn't have information about wheelchair accessibility. This is the **default** behavior. If the static GTFS contains a _wheelchair_accessible_ value, it won't be overwritten. |
+| **UNKNOWN** | The trip has no accessibility value present. This value will overwrite the value from the GTFS.  |
```

Comment:
Is this really required? Why not keep it simple with NO_VALUE as UNKNOWN?

Reply:
For example, if a static GTFS says trip A is wheelchair accessible, how can a trip_update changes the value to Unknown? If we drop `NO_VALUE` we won't support this case. Maybe this is an edge case and in the eventuality this happens the producer can put `WHEELCHAIR_INACCESSIBLE` instead. I don't feel too strongly on either choices.

Reply:
This is definitely confusing at first glance, but the underlying issue is that protobufs don't allow enums to have a `null` value.  I believe this is the first enumeration in GTFS that we're overriding in GTFS-rt, and as a result we haven't had this issue before.

From https://developers.google.com/protocol-buffers/docs/javatutorial?csw=1 (emphasis mine):

>`optional:` the field may or may not be set. If an optional field value isn't set, a default value is used. For simple types, you can specify your own default value, as we've done for the phone number type in the example. **Otherwise, a system default is used: zero for numeric types**, the empty string for strings, false for bools. For embedded messages, the default value is always the "default instance" or "prototype" of the message, which has none of its fields set. Calling the accessor to get the value of an optional (or required) field which has not been explicitly set always returns that field's default value.

I think it's important to support the case where the GTFS says that the trip is wheelchair accessible, but then real-time information indicates that this is unknown for some reason (`UNKNOWN`) - for example, let's say an accessible vehicle was scheduled to run a trip, but then it broke down and was replaced with another vehicle, and dispatch doesn't know if that vehicle is wheelchair accessible.  However, it's also important to say that there is no real-time information, and therefore the GTFS value should be used (`NO_VALUE`), especially for backwards compatibility and GTFS-rt feeds that aren't providing this field.

@gcamp The [proto3 spec says](https://developers.google.com/protocol-buffers/docs/proto3#enum):

>* There must be a zero value, so that we can use 0 as a numeric default value.
* The zero value needs to be the first element, for compatibility with the proto2 semantics where the first enum value is always the default.

We're currently using `proto2` for GTFS-realtime, but I'm assuming we should plan for eventually moving to `proto3`. This means that `NO_VALUE` would need to be `0`, which makes this even more confusing, as the enum values in GTFS-rt wouldn't match GTFS.

I'm definitely supportive of trying simpler iterative implementations first especially if we have willing producers for them, but in this case because of the above I'm wondering if we should look at a simplified version of the carriage proposal instead.

Reply:
>Not the same numerical value compared the static GTFS, but since protobuf libraries would use the name of the value anyway I'm not sure this is a problem.

Good point - in code I assume most/all protobuf libraries producing and consuming libraries would reference the enumeration name, not the ID, so maybe that's not too big of a stumbling block in software development, just confusing to read in the spec itself.

>Would the carriage proposal fix those issues? 

Yeah, thinking more, it doesn't.  I was thinking that because `CarriageDescriptor` is new, we wouldn't have the overriding issue that we would with GTFS `trip.wheelchair_accessible`.  But, we'd still have to specify some behavior for interaction between the two.  Unless we make `CarriageDescriptor.WheelchairAccessible` required, it could be empty, in which case we have the same overriding behavior problem as the default would be chosen.

So in summary using values from 0 to 4 for `TripUpdate.WheelchairAccessible` enumeration seems like a reasonable approach.

Given some of the potential interaction issues with GTFS here, I'd like to see a working implementation before we vote on it.  Having a producer and consumer is a requirement for calling a GTFS vote for new fields, but in the past those requirements have been relaxed for GTFS-rt given the potential cost of an agency reverting a deployed protobuf extension to the canonical protobuf after the field is approved.  Instead, a field has been voted on an accepted as "experimental" (see [occupancy proposal](https://groups.google.com/d/msg/gtfs-realtime/_HtNTGp5LxM/86U3GGxDbngJ).  Due to the interactions with GTFS, though, this proposal seems potentially more complicated to me.